WordPress/src/main/java/org/wordpress/android/ui/activitylog/ActivityLogListFragment.kt

```diff
@@ -0,0 +1,166 @@
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Context
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.util.DiffUtil
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.activity_log_list_fragment.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.ActivityLogModel
+import org.wordpress.android.util.NetworkUtils
+import org.wordpress.android.util.WPSwipeToRefreshHelper.buildSwipeToRefreshHelper
+import org.wordpress.android.util.helpers.SwipeToRefreshHelper
+import org.wordpress.android.viewmodel.activitylog.ActivityLogViewModel
+import java.util.Calendar
+import javax.inject.Inject
+
+class ActivityLogListFragment : Fragment() {
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: ActivityLogViewModel
+    private lateinit var swipeToRefreshHelper: SwipeToRefreshHelper
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.activity_log_list_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        activityLogList.layoutManager = LinearLayoutManager(activity, LinearLayoutManager.VERTICAL, false)
+
+        swipeToRefreshHelper = buildSwipeToRefreshHelper(
+                activityLogPullToRefresh,
+                {
+                    if (NetworkUtils.checkConnection(activity)) {
+                        viewModel.pullToRefresh()
+                    } else {
+                        swipeToRefreshHelper.isRefreshing = false
+                    }
+                })
+
+        (activity?.application as WordPress).component()?.inject(this)
+
+        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ActivityLogViewModel::class.java)
+        viewModel.site = if (savedInstanceState == null) {
+            activity?.intent?.getSerializableExtra(WordPress.SITE) as SiteModel
+        } else {
+            savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
+        }
+
+        setupObservers()
+
+        viewModel.start()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putSerializable(WordPress.SITE, viewModel.site)
+        super.onSaveInstanceState(outState)
+    }
+
+    private fun setupObservers() {
+        viewModel.events.observe(this, Observer<List<ActivityLogModel>> {
+            reloadEvents()
+        })
+
+        viewModel.eventListStatus.observe(this, Observer<ActivityLogViewModel.ActivityLogListStatus> { listStatus ->
```

Comment:
What do you think about exposing something like `LiveData<LoadingState>` instead of `LiveData<ActivityLogListStatus>`. The `LoadingState` could have boolean fields `isRefreshing` and `showLoadMore`. This way you'd move some logic to the viewModel, where it could be tested.

Reply:
After the read-only version of AL is shipped I plan to refactor the ViewModels and use the new [ListState](https://github.com/wordpress-mobile/WordPress-Android/pull/7644), which should address your comments. We first need to merge that PR, though.